### PR TITLE
Speed up Parzen Estimator.

### PR DESCRIPTION
**Pull request by oda**
This speeds up parzen estimator roughly 3 times.

### CODE REVIEW

**Comment by sile** [comment]
Please merge `master` branch to fix CI failure (see #553 for further details).

**Comment by oda** [comment]
I also merged support-tensorflow2 to fix the CL failure.

**Comment by sile** [comment]
FYI: https://github.com/pfnet/optuna/pull/562 (support-tensorflow2 branch) was merged.

**Comment by sile** [comment]
@g-votte @toshihikoyanase Please review this PR and merge if you are satisfied.

File: optuna/samplers/tpe/parzen_estimator.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def _calculate(
             consider_endpoints,  # type: bool
             weights_func  # type: Callable[[int], ndarray]
     ):
-        # type: (...) -> Tuple[List[float], List[float], List[float]]
+        # type: (...) -> Tuple[ndarray, ndarray, ndarray]
```
Comment (toshihikoyanase) [inline optuna/samplers/tpe/parzen_estimator.py]: I have a question about the code which uses the result of this method.
It still applies `numpy.asarray` to the returned values as follows:

```python
        s_weights, s_mus, sigmas = ParzenEstimator._calculate(
            mus, low, high, parameters.consider_prior, parameters.prior_weight,
            parameters.consider_magic_clip, parameters.consider_endpoints, parameters.weights)
        self.weights = numpy.asarray(s_weights)
        self.mus = numpy.asarray(s_mus)
        self.sigmas = numpy.asarray(sigmas)
```
Do you think it's necessary? I think they are already `ndarray`.

Comment (toshihikoyanase) [reply optuna/samplers/tpe/parzen_estimator.py]: I made a follow-up PR (#600) to handle this issue.

**Pull request merged**